lib/archethic/contracts.ex

```diff
@@ -255,27 +255,37 @@ defmodule Archethic.Contracts do
 
   defp get_condition_constants(
          :inherit,
-         %Contract{constants: %Constants{contract: contract_constant}},
+         %Contract{
+           constants: %Constants{contract: contract_constant},
+           public_functions: public_functions,
```

Comment:
I think it could be easier to use if we have the visibility inside the function itself, hence we could use pattern matching on it and avoid the `Map.merge()`
something like 
```elixir
%Contract{
  functions: %{
    "hello/0" => %{ast: ast, args: args, visibility:  :private
  }
}
```